### PR TITLE
Bump envoy-gloo and cloud-builders

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description
 - Bump envoy-gloo to v1.26.5-patch1
    - This pulls in fixes for recently disclosed CVEs in Envoy
 - Bump cloud builders to v0.6.5
   - This pulls in a go version bump that fixes recently disclosed CVEs in Go

### CODE REVIEW

File: changelog/v1.16.0-beta15/bump_envoy_gloo.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  dependencyRepo: cloud-builders
+  dependencyTag: v0.6.5
+  description: >-
+    Use the latest cloud builders to match go version of builder to that of enterprise
+- type: DEPENDENCY_BUMP
+  dependencyOwner: solo-io
+  dependencyRepo: envoy-gloo
+  dependencyTag: v1.26.5-patch1
+  description: >-
+    Use the latest cloud builders to match go version of builder to that of enterprise
```
Comment (nfuden) [inline changelog/v1.16.0-beta15/bump_envoy_gloo.yaml:13]: lets link the envoy cve this fixes (also update description)

File: changelog/v1.16.0-beta15/bump_envoy_gloo.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+changelog:
+- type: DEPENDENCY_BUMP
+  dependencyOwner: solo-io
+  dependencyRepo: cloud-builders
+  dependencyTag: v0.6.5
+  description: >-
+    Use the latest cloud builders to match go version of builder to that of enterprise
```
Comment (nfuden) [inline changelog/v1.16.0-beta15/bump_envoy_gloo.yaml:7]: Lets link the go security announce that this addresses https://groups.google.com/g/golang-announce/c/iNNxDTCjZvo and the cve inside it  CVE-2023-39325

File: changelog/v1.16.0-beta15/bump_envoy_gloo.yaml (hunk 1 above)
Comment (ben-taussig-solo) [reply changelog/v1.16.0-beta15/bump_envoy_gloo.yaml]: Updated here: https://github.com/solo-io/gloo/pull/8787/commits/59f6966d6b96145e98550d6712b9aa9fc8db671f

File: changelog/v1.16.0-beta15/bump_envoy_gloo.yaml (hunk 2 above)
Comment (ben-taussig-solo) [reply changelog/v1.16.0-beta15/bump_envoy_gloo.yaml:7]: Updated here: https://github.com/solo-io/gloo/pull/8787/commits/59f6966d6b96145e98550d6712b9aa9fc8db671f

**Review by sheidkamp — APPROVED** [review]

**Pull request merged**